test/skimNtuple_HHbtag.cpp

```diff
@@ -1038,172 +1046,211 @@ int main (int argc, char** argv)
-		  else if (HHrewType == kC2scan)  HHweight = hhreweighter->getWeight(mHH, ct1, c2_rew);
-		  else if (HHrewType == kOverRew) HHweight = hhreweighter->getWeight(mHH, ct1, kl_rew, kt_rew, c2_rew, cg_rew, c2g_rew);
-
-		  theSmallTree.m_genMHH = mHH;
-		  theSmallTree.m_genCosth = ct1;
-
-		  // cout << " ........... GEN FINISHED ........... " << " evt=" << theBigTree.EventNumber << " run=" << theBigTree.RunNumber << " lumi=" << theBigTree.lumi << endl;
+		  vHardScatter1.SetPxPyPzE (theBigTree.genpart_px->at(idx1hs), theBigTree.genpart_py->at(idx1hs), theBigTree.genpart_pz->at(idx1hs), theBigTree.genpart_e->at(idx1hs));
+		  vHardScatter2.SetPxPyPzE (theBigTree.genpart_px->at(idx2hs), theBigTree.genpart_py->at(idx2hs), theBigTree.genpart_pz->at(idx2hs), theBigTree.genpart_e->at(idx2hs));
+		}
+	      else {
+		cout << "** WARNING: couldn't find 2 H->tautau gen dec prod " << idx1hs << " " << idx2hs << endl;
+	      }
+	      
+	      vH1.SetPxPyPzE (theBigTree.genpart_px->at(idx1), theBigTree.genpart_py->at(idx1), theBigTree.genpart_pz->at(idx1), theBigTree.genpart_e->at(idx1) );
+	      vH2.SetPxPyPzE (theBigTree.genpart_px->at(idx2), theBigTree.genpart_py->at(idx2), theBigTree.genpart_pz->at(idx2), theBigTree.genpart_e->at(idx2) );
+	      theSmallTree.m_genH1_pt = vH1.Pt();
+	      theSmallTree.m_genH1_eta = vH1.Eta();
+	      theSmallTree.m_genH1_phi = vH1.Phi();
+	      theSmallTree.m_genH1_e = vH1.E();
+	      theSmallTree.m_genH2_pt = vH2.Pt();
+	      theSmallTree.m_genH2_eta = vH2.Eta();
+	      theSmallTree.m_genH2_phi = vH2.Phi();
+	      theSmallTree.m_genH2_e = vH2.E();
+	      vSum = vH1+vH2;
+	      mHH = vSum.M();
+	      vH1.Boost(-vSum.BoostVector());
+	      ct1 = vH1.CosTheta();
+	      
+	      
+	      // FRA DEBUG - build gen b jets
+	      if (idx1hs_b != -1 && idx2hs_b != -1)
+		{
+		  vGenB1.SetPxPyPzE (theBigTree.genpart_px->at(idx1hs_b), theBigTree.genpart_py->at(idx1hs_b), theBigTree.genpart_pz->at(idx1hs_b), theBigTree.genpart_e->at(idx1hs_b) );
+		  vGenB2.SetPxPyPzE (theBigTree.genpart_px->at(idx2hs_b), theBigTree.genpart_py->at(idx2hs_b), theBigTree.genpart_pz->at(idx2hs_b), theBigTree.genpart_e->at(idx2hs_b) );
 		}
+	      else {
+		cout << "** WARNING: couldn't find 2 H->bb gen dec prod " << idx1hs_b << " " << idx2hs_b << endl;
+	      }
+	      
+	      if (HHrewType == kDiffRew)      HHweight = hhreweighter->getWeight(mHH, ct1);
+	      else if (HHrewType == kC2scan)  HHweight = hhreweighter->getWeight(mHH, ct1, c2_rew);
+	      else if (HHrewType == kOverRew) HHweight = hhreweighter->getWeight(mHH, ct1, kl_rew, kt_rew, c2_rew, cg_rew, c2g_rew);
+	      
+	      theSmallTree.m_genMHH = mHH;
+	      theSmallTree.m_genCosth = ct1;
+	      
+	      // cout << " ........... GEN FINISHED ........... " << " evt=" << theBigTree.EventNumber << " run=" << theBigTree.RunNumber << " lumi=" << theBigTree.lumi << endl;
+	    } // if isHHsignal
+
+
```

Comment:
It's not clear to me that anything above this comment actually changed apart from the indentation; am I missing something?

Reply:
you're right..